labs/a8.md

```diff
@@ -0,0 +1,736 @@
+unable to sign their false statements such that they can be verified with 
+Natoshi's published public key because they don't have Natoshi's private key, 
+and you can rest assured that no one will unduly influence your project while 
+you go into hiding.
+
+However, in this scheme, how do you prevent an adversary from publishing a fake 
+public key and claiming to be you (they can make valid signatures against that 
+fake public key)? Somehow, you need to "bootstrap" trust: someone would need to 
+verify your identity and publicly affirm that your public key actually 
+corresponds to you. We do this by means of certificates: a signed statement 
+claiming that a particular public key does, in fact, belong to who it claims to 
+belong to. Who signs this certificate? A certificate authority, someone we 
+trust to be responsible about verifying identities and issuing signatures. But 
+how do we know which CAs to trust, and how can we trust that a CA that claims 
+to be trustworthy actually is? They probably need a certificate as well. It 
+sounds like it might be turtles all the way down; however, the chain does end 
+somewhere: the so-called root of trust, the root CAs. These are the CAs whose 
+certificates are pre-installed by browsers and operating systems and therefore 
+intrinsically trusted, without any further certificates necessary. If a root CA 
+signs your certificate, we assume they've done the due diligence necessary to 
+be willing to risk their reputation by signing your certificate, and basically 
+take their word for it. This model, known as the "Web of Trust" is how network 
+security works today. Unfortunately, it isn't as reliable as we may have hoped: 
+some CAs are scummy and will sign anything for enough money, resulting in valid 
+certificates being issued for domains like microsoft.com and github.com to 
+entities who are very obviously not Microsoft or GitHub.[^badwosign]
+
+You might not realize it, but you use and rely on certificates and signatures 
+every day. Any time you see a green lock near the address bar of a website you 
+visit, you are accesssing the site over a TLS or HTTPS connection, and the data 
+being transferred between you and the website is encrypted. When your browser 
+connects to the website's server, it asks for the server's public key in order 
+to set up an encrypted connection, and the server's certificate in order to 
+verify its identity as the server authorized to serve the domain you have 
+requested. Your browser then validates the public key by verifiying the 
+signatures on the certificate. If someone is attempting to perform a 
+man-in-the-middle attack on you, this certificate verification step will fail, 
+because it should be unlikely that a trusted CA will have issued a signed cert 
+for your domain to an entity other than you. You will get a very intrusive 
+notification informing you of this fact, and it is a bad idea to ignore the 
+certificate verification failure notification.
```

Comment:
This could be condensed a lot. Your example could be about HTTPS certs, using malicious wifi routers as the threat model. Also a cool lab activity could be to say "pick your favorite HTTPS website and figure out which root CA you're trusting when you visit it."